tchannel/tornado/peer.py

```diff
@@ -727,6 +736,11 @@ def choose(self, hostport=None, blacklist=None):
         if hostport:
             return self._get_isolated(hostport)
 
-        return self.peer_heap.smallest_peer(
-            (lambda p: p.hostport not in blacklist and not p.is_ephemeral),
-        )
+        def is_valid_peer(p):
```

Comment:
Perhaps `should_send_requests`? Given the number of concerns here, "valid" is pretty vague.